internal/command/platform/status.go

```
@@ -2,31 +2,135 @@ package platform
+	Description string `json:"description,omitempty"`
+	Indicator   string `json:"indicator,omitempty"`
+}
 
-	return
+type incidentUpdates struct {
+	Body      string    `json:"body,omitempty"`
+	Status    string    `json:"status,omitempty"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+}
+
+type Incident struct {
+	ID              string          `json:"id,omitempty"`
+	Name            string          `json:"name,omitempty"`
+	Status          string          `json:"status,omitempty"`
+	CreatedAt       time.Time       `json:"port,omitempty"`
+	Impact          string          `json:"impact,omitempty"`
+	MonitoringAt    time.Time       `json:"monitoring_at,omitempty"`
+	PageID          string          `json:"page_id,omitempty"`
+	ResolvedAt      time.Time       `json:"resolved_at,omitempty"`
+	UpdatedAt       time.Time       `json:"updated_at,omitempty"`
+	IncidentUpdates incidentUpdates `json:"incident_updates,omitempty"`
+}
+
+type StatusPage struct {
+	Status Status
+	Page   Page
+}
+type Client struct {
+	baseUrl    *url.URL
+	httpClient *http.Client
+}
+
+func createClient(ctx context.Context, rawUrl string) (*Client, error) {
+	cleanedURL, _ := url.Parse(rawUrl)
+	logger := logger.MaybeFromContext(ctx)
+	httpClient, err := api.NewHTTPClient(logger, http.DefaultTransport)
+	if err != nil {
+		return nil, fmt.Errorf("can't setup HTTP client to %s: %w", rawUrl, err)
+	}
+
+	return &Client{
+		baseUrl:    cleanedURL,
+		httpClient: httpClient,
+	}, nil
 }
 
 func runStatus(ctx context.Context) error {
-	const url = "https://status.fly.io/"
+	const url = "https://status.fly.io"
+	var (
+		cfg               = config.FromContext(ctx)
+		getStatusEndpoint string
+		getStatusKind     = flag.FirstArg(ctx)
+	)
+
+	switch getStatusKind {
+	case "incidents":
+		getStatusEndpoint = "api/v2/incidents/unresolved.json"
+	case "maintenance":
+		getStatusEndpoint = "api/v2/scheduled-maintenances/active.json"
```

what do you think if it accepts `status` or empty string and return an error otherwise?

---

Do you mean for the kind part of the command so `fly platform status status` or `fly platform status ` and those including maintenance and incidents?

---

Yes. Although `fly platform status status` reads weird

---

The whole point is that `fly platform status unknown-word` should return an error, not the status.

---

ah ok gotcha